File: models/code_exec.go
package models

// CodeExecutaion - describe a code execution task
type CodeExecutaion struct {
	TaskID      string `json:"id"`
	CodeSnippet string `json:"codeSnippet"`
	Language    string `json:"language"`
	IsCompleted bool   `json:"isCompleted"`
	Output      string `json:"output"`
	Error       string `json:"error,omitempty"`
}

// CodeExecutaionRequest - descripe a code execution request
type CodeExecutaionRequest struct {
	Language     string `json:"language"`
	CodeSnippent string `json:"code"`
}

// CodeExecutaionResponse - descripe a code execution response
type CodeExecutaionResponse struct {
	Error  string `json:"error,omitempty"`
	Output string `json:"output,omitempty"`
}

File: ui/src/services/code_exec/CodeExec.ts
interface CodeExecution {
  id: string;
  codeSnippet: string;
  language: string;
  isCompleted: string;
  output: string;
  error: string;
}

interface CodeExecutionController {
  data: {
    task: CodeExecution;
  };
  execute: (language: string, code: string) => void;
  cleare: () => void;
}

declare var codeExec: CodeExecutionController;

export default codeExec;
